plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightService.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    /**
+     * Starts the FlightService by initializing the stream manager.
+     */
+    @SuppressWarnings("removal")
+    @Override
+    protected void doStart() {
+        try {
+            allocator = AccessController.doPrivileged((PrivilegedAction<BufferAllocator>) () -> new RootAllocator(Integer.MAX_VALUE));
+            serverComponents.setAllocator(allocator);
+            SslContextProvider sslContextProvider = ServerConfig.isSslEnabled()
+                ? new DefaultSslContextProvider(secureTransportSettingsProvider)
+                : null;
+            serverComponents.setSslContextProvider(sslContextProvider);
+            serverComponents.initComponents();
+            clientManager = new FlightClientManager(
+                allocator, // sharing the same allocator between server and client
+                serverComponents.clusterService,
+                sslContextProvider,
+                serverComponents.workerEventLoopGroup, // sharing the same worker ELG between server and client
+                threadPool,
+                client
+            );
+            initializeStreamManager(clientManager);
+            serverComponents.setFlightProducer(new NoOpFlightProducer());
+            serverComponents.start();
+
+        } catch (Exception e) {
+            logger.error("Failed to start Flight server", e);
+            doClose();
+            throw new RuntimeException("Failed to start Flight server", e);
+        }
+    }
+
+    /**
+     * Retrieves the FlightClientManager used by the FlightService.
+     * @return The FlightClientManager instance.
+     */
+    public FlightClientManager getFlightClientManager() {
+        return clientManager;
+    }
+
+    /**
+     * Retrieves the StreamManager used by the FlightService.
+     * @return The StreamManager instance.
+     */
+    public StreamManager getStreamManager() {
+        return streamManager;
+    }
+
+    /**
+     * Retrieves the bound address of the FlightService.
+     * @return The BoundTransportAddress instance.
+     */
+    public BoundTransportAddress getBoundAddress() {
+        return serverComponents.getBoundAddress();
+    }
+
+    @VisibleForTesting
+    SslContextProvider getSslContextProvider() {
+        return serverComponents.getSslContextProvider();
+    }
+
+    /**
+     * Stops the FlightService by closing the server components and network resources.
+     */
+    @Override
+    protected void doStop() {
+        try {
+            AutoCloseables.close(serverComponents, streamManager, clientManager, allocator);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * doStop() ensures all resources are cleaned up and resources are recreated on
+     * doStart()
+     */
+    @Override
+    protected void doClose() {
+        doStop();
+    }
+
+    private void initializeStreamManager(FlightClientManager clientManager) {
+        streamManager = null;
```

Comment:
Not implemented?

Reply:
its part of the next PR https://github.com/opensearch-project/OpenSearch/pull/17313

Reply:
Just remove those methods please, let's introduce things when implemented

Reply:
done